core/test/solver/bicgstab.cpp

```diff
@@ -64,10 +64,9 @@ class Bicgstab : public ::testing::Test {
           bicgstab_factory(
               Solver::build()
                   .with_criteria(
-                      gko::stop::Iteration::build().with_max_iters(3u).on(exec),
+                      gko::stop::Iteration::build().with_max_iters(3u),
                       gko::stop::ResidualNorm<value_type>::build()
-                          .with_reduction_factor(gko::remove_complex<T>{1e-6})
-                          .on(exec))
+                          .with_reduction_factor(gko::remove_complex<T>{1e-6}))
```

Comment:
Could you add a test to ensure that the previous behaviour works as well ? Otherwise, it breaks all existing code.

Reply:
done